frontend/packages/ux-editor/src/utils/dataModelUtils.ts

```diff
@@ -0,0 +1,152 @@
+  dataBindingName: string,
+  dataModelFields: DataModelFieldElement[],
+): number => {
+  if (componentType === ComponentType.RepeatingGroup) {
+    const element = dataModelFields.find(
+      (e: DataModelFieldElement) => e.dataBindingName === dataBindingName,
+    );
+    return element?.maxOccurs;
+  }
+
+  return undefined;
+};
+
+export const getXsdDataTypeFromDataModelFields = (
+  componentType: ComponentType,
+  dataBindingName: string,
+  dataModelFields: DataModelFieldElement[],
+): string | undefined => {
+  if (componentType === ComponentType.Datepicker) {
+    const element = dataModelFields.find(
+      (e: DataModelFieldElement) => e.dataBindingName === dataBindingName,
+    );
+    return element?.xsdValueType === 'DateTime' ? 'DateTime' : undefined;
+  }
+
+  return undefined;
+};
+
+const generalFilter = (element: DataModelFieldElement) =>
+  element.dataBindingName && element.maxOccurs <= 1;
+const repeatingGroupFilter = (element: DataModelFieldElement) =>
+  element.dataBindingName && element.maxOccurs > 1;
+const multipleAttachmentsFilter = (element: DataModelFieldElement) =>
+  element.dataBindingName && element.maxOccurs > 1 && element.xsdValueType === 'String';
+
+export const getDataModelFieldsFilter = (
+  componentType: ComponentType,
+  label: boolean,
+): ((element: DataModelFieldElement) => boolean) => {
+  switch (componentType) {
+    case ComponentType.RepeatingGroup:
+      return repeatingGroupFilter;
+    case ComponentType.FileUpload:
+    case ComponentType.FileUploadWithTag:
+      return label ? multipleAttachmentsFilter : generalFilter;
+    default:
+      return generalFilter;
+  }
+};
+
+export type DataModelField = {
+  value: string;
+  label: string;
+};
+
+export const filterDataModelFields = (
+  getDataModelFieldsFilter: (element: DataModelFieldElement) => boolean,
+  data: DataModelFieldElement[],
+): DataModelField[] => {
+  if (!data) {
+    return [];
+  }
+  return data.filter(getDataModelFieldsFilter).map((element) => ({
+    value: element.dataBindingName,
+    label: element.dataBindingName,
+  }));
+};
+
+type DataModelFields = {
+  componentType: ComponentType;
+  bindingKey: string;
+  dataModelMetaData: DataModelFieldElement[];
+};
+
+export const getDataModelFields = ({
+  componentType,
+  bindingKey,
+  dataModelMetaData,
+}: DataModelFields): DataModelField[] => {
+  const filter = getDataModelFieldsFilter(componentType, bindingKey === 'list');
+  return filterDataModelFields(filter, dataModelMetaData);
+};
+
+export type InternalBindingFormat = {
+  field: string | undefined;
+  dataType: string | undefined;
+};
+
+export const convertDataBindingToInternalFormat = (
+  component: FormItem,
+  bindingKey: string,
+): InternalBindingFormat => {
+  const dataModelBinding =
+    component?.dataModelBindings && bindingKey in component.dataModelBindings
+      ? component.dataModelBindings[bindingKey]
+      : undefined;
+
+  const isOldOrNotSetFormat =
```

Comment:
Not sure, but is this a better name?
```suggestion
  const isOldFormatOrNotSet =
```